blockstore/apps/bundles/tests/test_store.py

```diff
@@ -88,12 +112,14 @@ def test_basic_creation(self, snapshot_created_mock):
                     public=False,
                     size=25,
                     hash_digest=create_hash(HTML_CONTENT_BYTES).digest(),
+                    mime_type='text/html',
                 ),
                 'test.txt': FileInfo(
                     path='test.txt',
                     public=False,
                     size=17,
                     hash_digest=create_hash(TEXT_CONTENT_BYTES).digest(),
+                    mime_type='text/plain',
```

Comment:
@jvdm, do you mind adding a file (or two) here that fall under the situation [Braden originally described](https://github.com/edx/blockstore/issues/63#issuecomment-893870623) where "two files are uploaded with different extensions but the same content"?  In particular, when such files resolve to the exact same MIME type.

If for nothing else other than to document that this was foreseen, and to protect this particular case from future bugs.  I couldn't detect a related problem just by reading the code.

Reply:
@arbrandes 

> do you mind adding a file (or two) here that fall under the situation Braden originally described where "two files are uploaded with different extensions but the same content"?

Can do. Although, I want to clarify what will happen.

> In particular, when such files resolve to the exact same MIME type.

Two scenarios, (a.) If the different extensions are in fact the same MIME, the files will dedup in the storage layer, just like today. That is correct because the extensions are the source of truth for the MIME-type header in the patch proposed, so one potential example is `cat.jpeg` and `samecat.jpg` will be stored as `<some-hash>.jpg`. Now, (b.) if different extensions are guessed to be two different MIME types, for example, if you do `mv cat.jpg cat.pdf` and upload `cat.pdf` and `cat.jpeg`, blockstore will create two files with the same content.

You might argue that for (b.), we would be better using the files' content as the source of the truth for the MIME header. That is possible, but in my opinion, not as flexible as using the file extension. One example, what if the content-based MIME type guessing fails for some (I know, very particular) case? Customers will feel confused, as in "Why my PDF is opening as an image?". And without a clear path to move forward. So, unless we add some additional API parameters to bypass the guessing, I prefer the file extension. Let me know if you want to explore the API parameter to select the type forcibly.

Reply:
For what it's worth, I think what you're describing in cases (a) and (b) sound good and I was thinking along the same lines too.